src/components/ItemList.svelte

```diff
@@ -28,39 +28,46 @@
   let currentPage = writable(1);
   setContext("currentPage", currentPage);
 
+  // track change of filter input
+  let search = $pageState.search || "";
+
   // re-filter items when showExpired or search text changes
   $: {
-    const search = $pageState.search || "";
-    // don't use $currentPage, because we don't want to call this reactively
-    // when paginating
-    currentPage.set(1);
-
-    const originMatch = (item) =>
-      item.origin && item.origin.includes(search.toLowerCase());
-
-    // filter on match either on name or on origin
-    filteredItems = items.filter(
-      (item) => item.name.includes(search) || originMatch(item)
-    );
-
-    // also filter out expired items (if we're not showing expired)
-    filteredItems = $pageState.showExpired
-      ? filteredItems
-      : filteredItems.filter((item) => !isExpired(item.expires));
+    if (search !== $pageState.search) {
+      search = $pageState.search || "";
+      // don't use $currentPage, because we don't want to call this reactively
+      // when paginating
+      currentPage.set(1);
+      $pageState.page = 1;
+      const originMatch = (item) =>
+        item.origin && item.origin.includes(search.toLowerCase());
+
+      // filter on match either on name or on origin
+      filteredItems = items.filter(
+        (item) => item.name.includes(search) || originMatch(item)
+      );
+
+      // also filter out expired items (if we're not showing expired)
+      filteredItems = $pageState.showExpired
+        ? filteredItems
+        : filteredItems.filter((item) => !isExpired(item.expires));
+    }
   }
 
   // update pagination when either pagination changes or filtered list changes
   // (above)
   $: {
+    $pageState.page = $currentPage || 1;
     const perPage = paginated ? DEFAULT_ITEMS_PER_PAGE : filteredItems.length;
+    const chunkIndex = paginated ? $currentPage - 1 : 0;
     pagedItems =
       filteredItems.length > 0
-        ? chunk([...filteredItems], perPage)[$currentPage - 1]
+        ? chunk([...filteredItems], perPage)[chunkIndex]
         : [];
   }
 
   const originClicked = (origin) => {
-    $pageState = { ...$pageState, search: origin };
+    $pageState = { ...$pageState, page: origin, search: origin };
```

Comment:
Hmm this doesn't look right. `origin` in this case is the Glean label while page is supposed to be our current page number, which is already set to 1 when search changes in the reactive `$` block above.

```suggestion
    $pageState = { ...$pageState, search: origin };
```